Dockerfile.spdk

```diff
@@ -0,0 +1,85 @@
+# syntax = docker/dockerfile:1.4
+ARG CEPH_SPDK_BASE="ubi"
+FROM quay.io/centos/centos:stream9 AS build
+
+ARG CEPH_VERSION
+RUN [ ! -z "${CEPH_VERSION}" ]
+
+COPY <<EOF /etc/yum.repos.d/ceph.repo 
+[Ceph]
+name=Ceph packages for \$basearch
+baseurl=https://download.ceph.com/rpm-$CEPH_VERSION/el\$releasever/\$basearch
+enabled=1
+priority=2
+gpgcheck=1
+gpgkey=https://download.ceph.com/keys/release.asc
+
+[Ceph-noarch]
+name=Ceph noarch packages
+baseurl=https://download.ceph.com/rpm-$CEPH_VERSION/el\$releasever/noarch
+enabled=1
+priority=2
+gpgcheck=1
+gpgkey=https://download.ceph.com/keys/release.asc
+EOF
+
+ARG SPDK_VERSION
+RUN [ ! -z "${SPDK_VERSION}" ]
+
+ARG PKGDEP_ARGS
+ARG CONFIGURE_ARGS
+ARG MAKEFLAGS
+ARG RPM_RELEASE="0"
+
+WORKDIR /src
+COPY . .
+
+RUN \
+    --mount=type=cache,target=/var/cache/dnf \
+    --mount=type=cache,target=/var/lib/dnf \
+    --mount=type=cache,target=/root/.cache/pip \
+    dnf install -y 'dnf-command(config-manager)' \
+    && dnf config-manager --set-enabled crb \
+    && dnf install -y \
+        rpm-build \
+        git-core \
+    && scripts/pkgdep.sh $PKGDEP_ARGS
+
+RUN DEPS="no" rpmbuild/rpm.sh $CONFIGURE_ARGS
+
+#------------------------------------------------------------------------------
+FROM registry.access.redhat.com/ubi9/ubi AS ceph-spdk-ubi
+
+ARG CENTOS_BASE="https://mirror.stream.centos.org/9-stream/BaseOS/x86_64/os/Packages/"
```

Comment:
One possible approach to avoid hardcoding `x86_64` could be to consider the following solution:
```suggestion
# Retrieve the machine architecture
RUN ARCH=$(uname -m)

ARG CENTOS_BASE="https://mirror.stream.centos.org/9-stream/BaseOS/$ARCH/os/Packages/"
```